File: JavaScript/프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    var days = [];
    for (let i=0; i<speeds.length; i++) {
        let day = Math.ceil((100 - progresses[i]) / speeds[i]);
        days.push(day);
    }
    
    while (days.length != 0) {
        let temp = days.splice(0,1);
        let count = 1;
        for (let day of days) {
            if (day > temp) break;
            count += 1;
        }
        answer.push(count);
        days.splice(0, count-1);
    }
    return answer;
}

File: Java/프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> a = new ArrayList<>();
        List<Integer> days = new ArrayList<>();
        
        for (int i=0; i<speeds.length; i++) {
            Double d = Math.ceil((100.0 - progresses[i]) / speeds[i]);
            int div = d.intValue();
            days.add(div);
        }
        
        while (!days.isEmpty()) {
            int temp = days.remove(0);
            int count = 1;
            for (int day : days) {
                if (day > temp) break;
                count += 1;
            }
            a.add(count);
            for (int i=1; i<count; i++) days.remove(0);
        }
        
        int[] answer = new int[a.size()];
        for (int i=0; i<answer.length; i++) {
            answer[i] = a.get(i);
        }
        return answer;
    }
}